### PR TITLE
feat(cli): config handler and prompter

### PR DESCRIPTION
**Pull request by karan-andalusia**

### CODE REVIEW

File: crates/cli/src/config/models/mod.rs [hunk 1]
```diff
@@ -1,30 +1,69 @@
...
 
-#[derive(Serialize, Deserialize, Debug)]
-pub struct Config {
-    pub version: ConfigVersion,
-    pub chain: Option<Chain>,
-    pub keystore: Keystore,
+use crate::types::Url;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Profile {
```
[the remark this answers is not in the document]
Comment (karan-andalusia) [reply crates/cli/src/config/models/mod.rs:11]: yeah config is a collection of multiple profiles

File: crates/cli/src/config/models/mod.rs [hunk 2]
```diff
@@ -1,30 +1,69 @@
...
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-pub enum ConfigVersion {
-    #[serde(rename = "v0")]
-    V0,
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Config {
+    #[serde(flatten)]
+    pub profiles: HashMap<String, Profile>,
```
[the remark this answers is not in the document]
Comment (krustaceany) [reply crates/cli/src/config/models/mod.rs:68]: yeah, @karan-andalusia please check for semver violations with `cargo semver-checks`

File: crates/cli/src/config/processor/prompt.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    if let Some(profile) = profile {
+        let chain = prompt_chain(Some(profile.chain));
+        let key_generation_folder = PathBuf::from(prompter::input::<String>(
+            "Enter key generation folder",
+            Some(profile.key_generation_folder.to_str().unwrap().to_string()),
+            None,
+        ))
+        .canonicalize()
+        .unwrap_or_else(|e| {
+            println!("{}", e.to_string().red());
```
Comment (krustaceany) [inline crates/cli/src/config/processor/prompt.rs:23]: can we do `e.red()` directly?

File: crates/cli/src/operator/processor/vault.rs [hunk 4]
```diff
@@ -79,22 +78,18 @@ pub async fn process_vault_creation<T: Transport + Clone, P: Provider<T> + Clone
...
-            .with_initial_text("0x")
-            .default(Bytes::default())
-            .allow_empty(true)
-            .interact()?;
+        let name = prompter::input("Please enter vault name", Some(asset_name), None);
+
+        let symbol = prompter::input("Please enter vault symbol", Some(asset_symbol), None);
+
+        let extra_data = prompter::input(
+            "Please enter any extra data",
```
Comment (krustaceany) [inline crates/cli/src/operator/processor/vault.rs:86]: nit: keep the colons

File: crates/cli/src/config/processor/prompt.rs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        // unwrap is safe since default is Some according to the if condition
+        match default.unwrap() {
+            Chain::Evm { id, rpc_url } => {
+                return Ok(Chain::Evm {
+                    id: prompter::input::<u64>("Enter chain ID", Some(id), None)?,
+                    rpc_url: prompter::input::<Url>("Enter RPC URL", Some(rpc_url), None)?,
+                })
+            }
+        }
+    }
```
Comment (krustaceany) [inline crates/cli/src/config/processor/prompt.rs:81]:
```suggestion
    if is_default {
        if let Some(Chain::Evm { id, rpc_url }) = default {
            return Ok(Chain::Evm {
                id: prompter::input::<u64>("Enter chain ID", Some(id), None)?,
                rpc_url: prompter::input::<Url>("Enter RPC URL", Some(rpc_url), None)?,
            });
        }
    }
```

File: crates/cli/src/operator/processor/mod.rs [hunk 6]
```diff
@@ -1,58 +1,92 @@
...
+                };
+
+                Keystore::Local {
+                    path: secp256k1_keystore_path,
+                }
+            }
+            // TODO: Update config to handle AWS secret and access keys
+            Keystore::Aws { secret: s } => Keystore::Aws { secret: s },
+        }
+    };
```
Comment (krustaceany) [inline crates/cli/src/operator/processor/mod.rs:57]:
```suggestion
    let secp256k1_keystore_type =
        if let Some(secp256k1_keystore_type) = args.secp256k1_keystore_type {
            match secp256k1_keystore_type {
                Keystore::Local { path: _ } => {
                    let secp256k1_keystore_path = match args.secp256k1_keystore_path {
                        Some(path) => path,
                        None => prompt_keystore_path()?,
                    };

                    Keystore::Local {
                        path: secp256k1_keystore_path,
                    }
                }
                // TODO: Update config to handle AWS secret and access keys
                Keystore::Aws { secret: s } => Keystore::Aws { secret: s },
            }
        } else {
            prompt_keystore_type(
                Curve::Secp256k1,
                profile.clone(),
                profile_name,
                config_path.clone(),
            )
            .await?
        };
```

File: crates/cli/src/operator/processor/mod.rs [hunk 7]
```diff
@@ -84,14 +118,54 @@ pub async fn process(args: OperatorArgs) -> eyre::Result<()> {
             dss_address,
             message,
             message_encoding,
-            core_address,
         } => {
-            let core_instance = CoreInstance::new(core_address, provider.clone());
+            let core_instance = CoreInstance::new(profile.core_address, provider.clone());
+
+            let bn254_keystore = if bn254_keystore.is_none() {
```
Comment (krustaceany) [inline crates/cli/src/operator/processor/mod.rs:124]: you can also refactor this like the one above

File: crates/cli/src/config/processor/prompt.rs (hunk 3 above)
Comment (karan-andalusia) [reply crates/cli/src/config/processor/prompt.rs:24]: sg, changed everywhere

File: crates/cli/src/operator/processor/mod.rs [hunk 8]
```diff
@@ -84,14 +119,54 @@ pub async fn process(args: OperatorArgs) -> eyre::Result<()> {
...
+                        };
+
+                        Keystore::Local {
+                            path: bn254_keypair_location,
+                        }
+                    }
+                    // TODO: Update config to handle AWS secret and access keys
+                    Keystore::Aws { secret: s } => Keystore::Aws { secret: s },
+                }
+            };
```
Comment (krustaceany) [inline crates/cli/src/operator/processor/mod.rs:148]:
```suggestion
            let bn254_keystore = match bn254_keystore {
                Some(Keystore::Local { path: _ }) => {
                    let bn254_keypair_location = match bn254_keypair_location {
                        Some(path) => path,
                        None => prompt_keystore_path()?,
                    };

                    Keystore::Local {
                        path: bn254_keypair_location,
                    }
                }
                // TODO: Update config to handle AWS secret and access keys
                Some(Keystore::Aws { secret: s }) => Keystore::Aws { secret: s },
                None => {
                    prompt_keystore_type(
                        Curve::Bn254,
                        profile.clone(),
                        profile_name,
                        config_path.clone(),
                    )
                    .await?
                }
            };
```

File: crates/cli/src/operator/processor/mod.rs [hunk 9]
```diff
@@ -1,58 +1,93 @@
...
+            }
+        } else {
+            prompt_keystore_type(
+                Curve::Secp256k1,
+                profile.clone(),
+                profile_name,
+                config_path.clone(),
+            )
+            .await?
+        };
```
Comment (krustaceany) [inline crates/cli/src/operator/processor/mod.rs:58]:
```suggestion
    let secp256k1_keystore_type = match args.secp256k1_keystore_type {
        Some(Keystore::Local { path: _ }) => {
            let secp256k1_keystore_path = match args.secp256k1_keystore_path {
                Some(path) => path,
                None => prompt_keystore_path()?,
            };

            Keystore::Local {
                path: secp256k1_keystore_path,
            }
        }
        // TODO: Update config to handle AWS secret and access keys
        Some(Keystore::Aws { secret: s }) => Keystore::Aws { secret: s },
        None => {
            prompt_keystore_type(
                Curve::Secp256k1,
                profile.clone(),
                profile_name,
                config_path.clone(),
            )
            .await?
        }
    };
```

**Pull request merged**